### DISCUSSION TITLE
Automatic deletion/expiration of documents after x years/days

### DISCUSSION THREAD

**Discussion by holzhannes** [jonaswinkler/paperless-ng#733] [Feature Requests]
Idea: It would be nice to have a meta value `expiration-date` for each document which could be prefilled and updated based on the `issued date` + a value of `x years`.

How many years could be a global value or specified by the document type. Thus, it would be easy to automatically delete documents after the legal retention periods of e.g. 3, 6 or 10 years, especially if the documents contain personal data. It would be ingenious if the x years could also be related to the end of the year of the document's issue date. 

"Expiration after 10 years at the end of the year".
e.g. document dated 08/02/2020 should be deleted after 10 years at the end of the year i.e. 01/01/2031

**Comment by rauferd** [comment]
This, in my opinion, is an awesome idea.

The whole purpose of document management is to keep information available and accessible for as long as _necessary_. For many documents "as long as necessary" does not mean indefinitely. As already pointed out, some documents are even legally required to be deleted after a certain amount of time.

**Reply by harmlessdonkey** [reply]
Wouldn't even need to be automatically. I would like a field with an expiration date that I could query from time-to-time to find documents past their expiration date. My main issue is not deleting the digital copy (but others have made good point as to why this is important), but rather creating a workflow to shred the physical copy of the document after a period of time.

**Reply by rauferd** [reply]
I second that!

While beyond the scope of automatic deletion (originally intended), this would be a great new feature:
 Many offices work with a system called "Wiedervorlage" in German (is "bring forward file system" the correct English translation?). This means that files are organized by their due date and pop up "automatically", when someone needs to work on them.
Due dates could be assigned manually or even automatically by paperless-ng (e.g. immediately for incoming bills; by the end of the quarter for tax documents; end of the year for insurace contracts, etc. - this seems to be a quite suitable task for machine learning - or - the due date could be related to a tag or a combination of tags that paperless-ng can already assign automatically).

One could then create a view for the currently due files.
When idle, people can work on documents with due dates in the future (another view).

Awesome idea, @harmlessdonkey. :+1:

**Reply by BerFoe** [reply]
I would like to highlight this feature request again and also elaborate on it a bit, because it is absolutely essential, especially for companies in Germany. In addition, in my eyes, this is also one of only two missing requirements imposed by the legislature in Germany.
To continue the topic a bit further: We need to distinguish between a minimum retention period (may be deleted in X years/months) and a maximum retention period (must be deleted after X years/months). The former applies to almost all conventional documents, e.g. it is ten years for invoices. The maximum retention period, on the other hand, is important for all documents that are relevant from a data protection perspective. For example, application documents must be deleted after six months at the latest.

**Comment by tomdeck** [comment]
I came here to ask for the same feature. I think it would also make sense to make a distinction between the retention period of the paper version (after which, physical action is needed: removing / shredding the document) and the retention period of the scanned version (after which the file could be deleted even automatically).

**Comment by iverona** [comment]
Hi,

if anyone is interested on this feature please check: https://github.com/jonaswinkler/paperless-ng/discussions/1593

I'll be happy to work with others to improve it, as it's really basic in the current state.
